### ISSUE TITLE
[expo-notifications] Android notification icon when using useNextNotificationsApi

### ISSUE THREAD

**Issue by 1337LutZ** [expo/expo#9062]
### Steps to Reproduce

1. Setup a bare Expo project
2. Add the PN example from the Expo docs
3. Switch to the new `useNextNotificationsApi` code.
4. Send a PN on your Android device

_I cannot create a snack as it doesn't seem to allow for modifying the `app.json`._

Repository: https://github.com/1337LutZ/android-pn-icon-bug

### Expected Behavior vs Actual Behavior
It used to work as intended:
- Transparency shows as transparant
- Anything white will be opaque in the bar / notification list. (white or chosen icon color)

Actual behavior:
- Transparency inside an icon now no longer gets "cut out". The icon will show only as a white/black square/circle depending on your OS settings.

Screenshots:
![Screenshot_20200702-092217](https://user-images.githubusercontent.com/1216787/86329333-4899e800-bc46-11ea-8499-5cb545c729df.png)
![Screenshot_20200702-092227](https://user-images.githubusercontent.com/1216787/86329335-49327e80-bc46-11ea-8185-a75f52d8253a.png)

**Comment by neelrocketbots** [comment]
Ya, experiencing the same thing as well. After upgrading from SDK 37 to 38, the icon is no longer showing and in it's place, my users are getting a white square icon on push notification. After removing the `useNextNotificationsApi` in app.json, I got the icon working fine again.

**Comment by ksrudeboy** [comment]
Not only SDK 38 but it is intermittently showing sometimes the same behaviour in SDK 37. I reported this yesterday. 
https://github.com/expo/expo/issues/9048

**Comment by 1337LutZ** [comment]
> Not only SDK 38 but it is intermittently showing sometimes the same behaviour in SDK 37. I reported this yesterday.
> #9048
> You can see all screenshot and example of another app also.

Can you reproduce this using my repository and provide screenshots?

**Comment by ksrudeboy** [comment]
Sure

**Comment by RodolfoGS** [comment]
I could reproduce your issue, also I notice that the notification color also have problems.

**useNextNotificationsApi: true**
 - the icon is square
 - color isn't respected

**useNextNotificationsApi: false**
 - the icon is not respected, appear the Expo default icon
 - color works great

Color that I used for both tests: `"color": "#800080"`

![image](https://user-images.githubusercontent.com/870365/86360781-7bd57a80-bc38-11ea-8466-1a235f1283da.png)

Note: I tested with an icon that works in another app, is a gray scale icon, square, 96x96

**Comment by edwardkong86** [comment]
After I upgrage from SDK36 to 38, I encounter the notification icon and app title is missing from the Notification Popup when build as standalone Android apps

SDK:38

Android Version 10.0
![WhatsApp Image 2020-07-03 at 5 34 44 PM](https://user-images.githubusercontent.com/12832857/86455941-f1823900-bd53-11ea-87e1-c9e3b66db3b1.jpeg)

Android Version 6.0.1
![WhatsApp Image 2020-07-03 at 5 35 02 PM](https://user-images.githubusercontent.com/12832857/86455938-ef1fdf00-bd53-11ea-94a8-40bd8e0deefe.jpeg)

**Comment by tr-joe** [comment]
Experiencing the Same Issue as mentioned Above. Please see screenshot... Would really appreciate some fast update on this issue as we have this version in the App stores used by customers.

Thank you
Duplicate of #9062
![106162908_582341425988612_9005701957657800523_n](https://user-images.githubusercontent.com/48056786/86486061-41600080-bd52-11ea-8a90-ebde32a307d0.jpg)

**Comment by Aryk** [comment]
> Ya, experiencing the same thing as well. After upgrading from SDK 37 to 38, the icon is no longer showing and in it's place, my users are getting a white square icon on push notification. After removing the `useNextNotificationsApi` in app.json, I got the icon working fine again.

@neelrocketbots Are you putting it under expo.android.useNextNotificationsApi = true ? I'm doing that and I'm on SDK 38 and it says it is not a valid key.

**Comment by 1337LutZ** [comment]
> > Ya, experiencing the same thing as well. After upgrading from SDK 37 to 38, the icon is no longer showing and in it's place, my users are getting a white square icon on push notification. After removing the `useNextNotificationsApi` in app.json, I got the icon working fine again.
> 
> @neelrocketbots Are you putting it under expo.android.useNextNotificationsApi = true ? I'm doing that and I'm on SDK 38 and it says it is not a valid key.

I did it as well in SDK 38. Did you upgrade using expo upgrade?

**Comment by Aryk** [comment]
yes I did

**Comment by 1337LutZ** [comment]
Any update on this?

**Comment by tr-joe** [comment]
HI Dude, I saw on Monday @lukmccall  you picked it up mate and made a pull request that failed... Do you have any update on when this will be completed and can you confirm this will resolve the Issue of not being able to save Notification Icon and change color? Additionally, I also noticed that when you create an Android channel using new API although you set light color and other properties it just ignores them when I use the request to check the channel it only contains all default settings apart from the Priority , Name, Description changes.

**Comment by lukmccall** [comment]
Hi @tr-joe, the PR which you mentioned will be merged soon. However, I don't know when it will be published. 

By 

> when I use the request to check the channel

did you mean `Notifications.getChannelAsync()`? If yes, please create a new issue for that ;)

**Comment by 1337LutZ** [comment]
@lukmccall Was this fix already published?

**Comment by TheJoshE** [comment]
I was trying to fix my icon file for hours yesterday thinking it was a problem with the PNG type!

Do you have any idea when this will be published?

**Comment by 0plus1** [comment]
Reporting the same exact issue

**Comment by alex-fournier** [comment]
This is a very annoying issue, a quick release of the fix would be much appreciated.
Our production app is using push notifications a lot.

**Comment by 1337LutZ** [comment]
@byCedric Sorry for the tag, any eta on a publish for this fix?

**Comment by estitesc** [comment]
We are also waiting on this fix before doing our first release of our current project. Please + thank you.

**Comment by zencephalon** [comment]
+1 It would be really great to have this fix published.

**Comment by JoaRiski** [comment]
Also seeing the same issue

**Comment by YahyaBagia** [comment]
Any update on this?
I'm having the same issue. Notification Icon was showing fine on SDK 37, upgrading to SDK 38 turned it to a grey color filled circle.

**Comment by alex-fournier** [comment]
I tried `expo-notifications@0.5.0` (which includes https://github.com/expo/expo/pull/9116) with Expo SDK 38 but I'm still having the same issue :(

![icon](https://user-images.githubusercontent.com/1874622/88810667-f4871280-d1b5-11ea-8a12-db7382ad1d95.jpg)

**Comment by lukmccall** [comment]
@alex-fournier we didn't update builders. So, native changes won't be available if you're building an app using our servers.

**Comment by mitchobrian** [comment]
Is it a Bug in SDK38?

**Comment by lukmccall** [comment]
@mitchobrian, yes, this occurs in the `SDK38`.

**Comment by mitchobrian** [comment]
Are u on it, or could we help anyhow? Or should I downgrade?

Łukasz Kosmaty <notifications@github.com> schrieb am Do., 30. Juli 2020,
13:39:

**Comment by SleeplessByte** [comment]
Don't use the new notifications API until SDK 39, or when a build-service fix is done.

**Comment by mitchobrian** [comment]
> Don't use the new notifications API until SDK 39, or when a build-service fix is done.

Really? Do you have a roadmap for SDK39? I fixed it with a workaround: https://forums.expo.io/t/how-to-set-expo-notification-icon-sdk38-properly/40994/2

**Comment by SleeplessByte** [comment]
> > Don't use the new notifications API until SDK 39, or when a build-service fix is done.
> 
> Really? Do you have a roadmap for SDK39? I fixed it with a workaround: https://forums.expo.io/t/how-to-set-expo-notification-icon-sdk38-properly/40994/2

That workaround most likely kills your android app icon. If you have a adaptable icon set, then modern devices will work, probably, but still it will not be what you expect on those without adaptive support.

**Comment by ValentinH** [comment]
OMG I spent hours on this. Indeed, I started implementing `notifications` last week so I didn't know this was a bug with SDK 38. I thought that my assets were wrong. 😱

**Comment by ValentinH** [comment]
@lukmccall I'm not sure to understand how to get the fix (if we can get it now). I'm building locally with `turtle` and using SDK 38: do I need to wait for SDK 39 or it should be fixed if I use the latest `expo-notifications`?

**Comment by lukmccall** [comment]
@ValentinH, unfortunately, you need to wait for SDK 39. This fix is only available in the Bare workflow for now.

**Comment by ValentinH** [comment]
Thanks for the answer! It might be the time to eject for me :)

**Comment by IronYeti** [comment]
@ValentinH - I rolled back to SDK 37. There wasn't anything in 38 that I absolutely needed so this option worked for me.

**Comment by ValentinH** [comment]
Thanks @IronYeti . I do need SDK 38 because I need local notifications to be persisted across reboots on Android 🙂

**Comment by SleeplessByte** [comment]
FYI: You don't need to roll back to SDK 37 to use the previous notifications API.

**Comment by ValentinH** [comment]
According to https://github.com/expo/expo/issues/9062#issuecomment-652989917 there seems to also have issues with the previous API.
I tried it and couldn't see the icon. It might mean that there is also something wrong on my side.

Am I supposed to see my notification icon when using the Expo client ?

**Comment by 1337LutZ** [comment]
> According to [#9062 (comment)](https://github.com/expo/expo/issues/9062#issuecomment-652989917) there seems to also have issues with the previous API.
> I tried it and couldn't see the icon. It might mean that there is also something wrong on my side.
> 
> Am I supposed to see my notification icon when using the Expo client ?

When using the Expo client you will see the Expo icon, not your. Atleast from my experience.

**Comment by mitchobrian** [comment]
> FYI: You don't need to roll back to SDK 37 to use the previous notifications API.

Thanks, when using the old version, what is the equivalent to the listener Notifications.addNotificationReceivedListener in the older version?

**Comment by SleeplessByte** [comment]
> > According to [#9062 (comment)](https://github.com/expo/expo/issues/9062#issuecomment-652989917) there seems to also have issues with the previous API.
> > I tried it and couldn't see the icon. It might mean that there is also something wrong on my side.
> > Am I supposed to see my notification icon when using the Expo client ?
> 
> When using the Expo client you will see the Expo icon, not your. Atleast from my experience.

This is correct. That's because the notification icon needs to be _bundled_ into the binary.

**Comment by SleeplessByte** [comment]
> > FYI: You don't need to roll back to SDK 37 to use the previous notifications API.
> 
> Thanks, when using the old version, what is the equivalent to the listener Notifications.addNotificationReceivedListener in the older version?

https://docs.expo.io/versions/latest/sdk/legacy-notifications/#notificationsaddlistenerlistener

Yes this works when the app is running. We use this in production to show a dialog in-app. We purposefully show it _both_ as a system notification and in the app, because that works well for us, but in general this has worked and still works for the past years. It's not as fine-grained (or **perfect**) as the new API, but should be fine for now whilst they fix it.

**Comment by dmbejar** [comment]
In my case, it was working right with the old SDK and 'expo', but when I changed last month to latest SDK and 'expo-notifications' now instead the icon and color selected for icon at Android, it shows the app icon.

**Comment by mrfsbDev** [comment]
This new API displays the notification in a more friendly way, the old API stacks the notifications and it is impossible to view them, this icon problem makes it impossible to use.

**Comment by mrfsbDev** [comment]
The SDK 39 be released this month?

**Comment by lukmccall** [comment]
@mrfsbDev, the new SDK will be released soon ;)

**Comment by vandemberg** [comment]
The new SDK 39 resolve the problem <3

**Comment by Lorezz** [comment]
mmm... i have the same issue on sdk 39 on built apk, but only on some android os,  I'll try to play with colors and icons inside android  part  of  app.json

**Comment by ivansenic** [comment]
> mmm... i have the same issue on sdk 39 on built apk, but apparently only on some android os

Same here, this an example from the build app on OnePlus6.. Our icon is set as the notification media, and push notification icon is expo's one. Any ideas what could lead to this?

![Screenshot_20200926-095653__01.jpg](https://user-images.githubusercontent.com/10600041/94335963-a046b600-ffdf-11ea-8089-3134e75ff1f1.jpg)

**Comment by lukmccall** [comment]
@ivansenic, could you open a new issue for that?

**Comment by ivansenic** [comment]
> @ivansenic, could you open a new issue for that?

Sure, here it is -> #10469
